src/Surfnet/StepupGateway/GatewayBundle/Controller/GatewayController.php

```diff
@@ -174,10 +170,19 @@ public function sendAuthenticationCancelledByUser(): Response
         // @see: \Surfnet\StepupGateway\GatewayBundle\Controller\SecondFactorController::cancelAuthenticationAction
         $requestStack = $this->get('request_stack');
         $request = $requestStack->getParentRequest();
-        if (!$request->get('authenticationMode', false)) {
+
+        if ($request === null) {
+            throw new RuntimeException('No request');
+        }
+
+        $authenticationMode = $request->query->getString('authenticationMode');
+        if ($authenticationMode === '') {
+            $authenticationMode = $request->request->getString('authenticationMode');
+        }
```

Comment:
I dont think we'll need the fallback to the post request params. AFAIR the authentication mode is always passed as a query parameter.